src/header/header.scss

```diff
@@ -227,7 +226,7 @@
 
   gap: 0;
   max-width: 25vw;
-  height: 100%;
+  height: calc(100% - 2px); // `- 2px` accounts for border width
```

Comment:
I think I will create a variable like `border-width: 1px` to make these changes clear without the comments, and for use in all cases where border is 1px.